lib/matplotlib/backends/backend_wx.py

```diff
@@ -971,9 +971,14 @@ def _onClose(self, event):
         _log.debug("%s - onClose()", type(self))
         self.canvas.close_event()
         self.canvas.stop_event_loop()
-        Gcf.destroy(self)
-        if self:
-            self.Destroy()
+        # set FigureManagerWx.frame to None to prevent repeated attempts to
+        # close this frame from with the FigureManagerWx.destroy()
+        if self.figmgr is not None:
```

Comment:
I don't understand how this works, as I don't see anything setting `self.figmgr = None`.

Reply:
https://github.com/matplotlib/matplotlib/blob/fa73bc09d62aef99842bbbd13dca687ac5bbd0b9/lib/matplotlib/backends/backend_wx.py#L924

Reply:
That's not `None`...

Reply:
oh, sorry miss read your comment :sheep:

Reply:
My mistake - its completely unnecessary. Can just do:
```
def _onClose(self, event):
        _log.debug("%s - onClose()", type(self))
        self.canvas.close_event()
        self.canvas.stop_event_loop()
        # set FigureManagerWx.frame to None to prevent repeated attempts to close
        # this frame from with the FigureManagerWx.destroy()
        self.figmgr.frame = None 
        # remove figure manager from Gcf.figs
        Gcf.destroy(self.figmgr)
        #carry on with wx close event propagation, frame&children destruction
        event.Skip() 
```